sdk/python/kubeflow/trainer/hf_llm_training.py

```diff
@@ -28,17 +28,26 @@
 logger.setLevel(logging.INFO)
 
 
-def setup_model_and_tokenizer(model_uri, transformer_type, model_dir):
+def setup_model_and_tokenizer(model_uri, transformer_type, model_dir, num_labels):
     # Set up the model and tokenizer
     parsed_uri = urlparse(model_uri)
     model_name = parsed_uri.netloc + parsed_uri.path
 
-    model = transformer_type.from_pretrained(
-        pretrained_model_name_or_path=model_name,
-        cache_dir=model_dir,
-        local_files_only=True,
-        trust_remote_code=True,
-    )
+    if num_labels is not None and num_labels != "None":
```

Comment:
When num_labels can be equal to `"None"` ?
Isn't the `num_labels is not None` check is sufficient ?

Reply:
@andreyvelich When users don't set num_labels, it will be the string "None" instead of the actual None value due to a string transformation:
<img width="531" alt="image" src="https://github.com/user-attachments/assets/d643b116-06c4-4226-806d-02f7fe583607">
Should I remove the condition `num_labels is not None` to avoid confusion, or should I add a check to handle the "None" string before passing `num_labels` to `args`?

Reply:
In that case, just set the default value as "None" and make this single check here: https://github.com/kubeflow/training-operator/blob/56f112b9655ab14d4d272713f32c59a863cd6a03/sdk/python/kubeflow/trainer/hf_llm_training.py#L163

Reply:
That's good advice. Thank you! I have made the adjustment.